### PR TITLE
defaults for each contextvar

### PR DESCRIPTION
**Pull request by sfc-gh-pmardziel**
# Description

Prevents the contextvar LookupError.

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Set default values for `ContextVar` instances in multiple classes to prevent `LookupError`.
> 
>   - **Behavior**:
>     - Set default value `None` for `recording_contexts` in `App` class in `app.py` to prevent `LookupError`.
>     - Set default value `{}` for `_context_endpoints` in `EndpointSetup` class in `endpoint.py` to prevent `LookupError`.
>     - Set default values `set()` and `{}` for `_context_contexts` and `_stack_contexts` in `WithInstrumentCallbacks` class in `instruments.py` to prevent `LookupError`.
>     - Set default value `set()` for `formatted_objects` in `SerialModel` class in `serial.py` to prevent `LookupError`.
>   - **Misc**:
>     - Update `api.trulens.3.11.yaml` to reflect changes in the codebase.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for 4ef5de530df3294bf8a880469ac4f8ac8a782295. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by sfc-gh-dkurokawa — COMMENTED** [review]
I think there's another one in `serial.py`.

Also, doesn't this mean it won't work in colab and snowflake notebooks though?

File: tests/unit/static/golden/api.trulens.3.11.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ trulens.benchmark.test_cases:
     generate_summeval_groundedness_golden_set: builtins.function
 trulens.core:
   __class__: builtins.module
-  __version__: 1.1.0
+  __version__: 1.0.2a0
```
Comment (sfc-gh-jreini) [inline tests/unit/static/golden/api.trulens.3.11.yaml:63]: bad merge?

Comment (sfc-gh-pmardziel) [reply tests/unit/static/golden/api.trulens.3.11.yaml:63]: `1.0.2a0` seems to be accurate relative to what pyproject.toml states on main. What is it supposed to be?

Comment (sfc-gh-pmardziel) [reply tests/unit/static/golden/api.trulens.3.11.yaml:63]: https://github.com/truera/trulens/pull/1539 probably needs to be merged in?

Comment (sfc-gh-jreini) [reply tests/unit/static/golden/api.trulens.3.11.yaml:63]: Ahh - just merged it

**Review by sfc-gh-dkurokawa — APPROVED** [review]

**Pull request merged**